### PR TITLE
[FIX] mail: speed up update cycle and create/insert

### PR DESCRIPTION
**Pull request by seb-odoo**
Part of task-2308312

### CODE REVIEW

File: addons/mail/static/src/model/model_manager.js [hunk 1]
```diff
@@ -115,46 +115,56 @@ class ModelManager {
      * from this record creation.
      *
      * @param {mail.model} Model class
-     * @param {Object} [data={}]
+     * @param {Object|Object[]} [data={}]
      * @returns {mail.model} newly created record
```
Comment (alexkuhn) [inline addons/mail/static/src/model/model_manager.js:119]: could return `mail.model[]`

File: addons/mail/static/src/model/model_manager.js [hunk 2]
```diff
@@ -253,17 +267,27 @@ class ModelManager {
      * per "unique find" criteria from data on Model.
      *
      * @param {mail.model} Model class
-     * @param {Object} data
+     * @param {Object|Object[]} data
      * @returns {mail.model} created or updated record.
```
Comment (alexkuhn) [inline addons/mail/static/src/model/model_manager.js:271]: same, `|mail.model[]`

File: addons/mail/static/src/model/model_manager.js [hunk 3]
```diff
@@ -740,7 +737,10 @@ class ModelManager {
                 if (k === 'constructor') {
                     return target[k];
                 }
-                const field = target.constructor.fields[k];
+                const fields = target.constructor.fields;
+                const field = Object.prototype.hasOwnProperty.call(fields, k)
+                    ? fields[k]
+                    : undefined;
```
Comment (alexkuhn) [inline addons/mail/static/src/model/model_manager.js:743]: weird, all models have `.fields`, no?

File: addons/mail/static/src/model/model_manager.js [hunk 4]
```diff
@@ -917,6 +917,46 @@ class ModelManager {
         }
     }
 
+    /**
+     * Executes the provided function as part of a single update cycle. This
+     * allows the execution of computed fields to happen only once, at the end
+     * of the last pending update cycle.
+     *
+     * @private
+     * @param {function} func
```
Comment (alexkuhn) [inline addons/mail/static/src/model/model_manager.js:926]: would clarify this function is intended to make changes by mean of `.update()`, just so it's easier to understand why it makes sense to wrap this function.

File: addons/mail/static/src/models/model/model.js [hunk 5]
```diff
@@ -61,7 +61,7 @@ function factory() {
          * function, otherwise the record will not be registered.
          *
          * @static
-         * @param {Object} [data] data object with initial data, including relations.
+         * @param {Object|Object[]} [data] data object with initial data, including relations.
          * @returns {mail.model} newly created record
```
Comment (alexkuhn) [inline addons/mail/static/src/models/model/model.js:65]: `|mail.model[]`

File: addons/mail/static/src/models/model/model.js [hunk 6]
```diff
@@ -99,7 +99,7 @@ function factory() {
          * on provided data.
          *
          * @static
-         * @param {Object} data
+         * @param {Object|Object[]} data
          * @returns {mail.model} created or updated record.
```
Comment (alexkuhn) [inline addons/mail/static/src/models/model/model.js:103]: `|mail.model[]`

File: addons/mail/static/src/model/model_manager.js (hunk 3 above)
Comment (seb-odoo) [reply addons/mail/static/src/model/model_manager.js:745]: to handle `toString` and other magic/inherited properties that are not fields

**Pull request closed without merging**